docs/library/machine.CAN.rst

```diff
@@ -0,0 +1,476 @@
+.. currentmodule:: machine
+.. _machine.CAN:
+
+class CAN -- Controller Area Network protocol
+=============================================
+
+CAN is a two-wire serial protocol used for reliable real-time message delivery
+between one or more nodes connected to a common bus. CAN 2.0 was standardised in
+ISO-11898, and is now also known as CAN Classic.
+
+There is also a newer, backwards compatible, protocol named CAN FD (CAN with
+Flexible Data-Rate).
+
+CAN support requires a controller (often an internal microcontroller
+peripheral), and an external transceiver to level-shift the signals onto the CAN
+bus.
+
+The ``machine.CAN`` interface is a *low level basic* CAN messaging interface
+that abstracts a CAN controller as an outgoing priority queue for sending
+messages, an incoming queue for receiving messages, and mechanisms for reporting
+errors.
+
+.. note:: The planned ``can`` and ``aiocan`` micropython-lib modules will be the
+          recommended way to use CAN with MicroPython.
+
+Constructor
+-----------
+
+.. class:: CAN(id, bitrate, *args, **kwargs)
+
+   Construct a CAN controller object of the given id:
+
+   - ``id`` identifies a particular CAN controller object; it is board and port specific.
+   - ``bitrate`` is the initial bus bit rate in bits per second.
```

Comment:
Why is `bitrate` explicitly listed here, and then also part of `init()`?  Would it be clearer to just leave it out of the argument list here?